embedded-graphics/src/image/mod.rs

```diff
@@ -1,33 +1,262 @@
+///
+/// [module documentation]: ./index.html
+#[derive(Debug, Clone, Copy)]
+pub struct Image<'a, I, C> {
+    image_data: &'a I,
+    offset: Point,
+    c: PhantomData<C>,
+}
+
+impl<'a, I, C> Image<'a, I, C>
+where
+    &'a I: IntoPixelIter<C>,
+    I: ImageDimensions,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Create a new `Image` with a given [`IntoPixelIter`]
+    ///
+    /// The passed [`IntoPixelIter`] provides a source of pixel data from the original image.
+    ///
+    /// [`IntoPixelIter`]: ./trait.IntoPixelIter.html
+    pub fn new(image_data: &'a I, position: Point) -> Self {
+        Self {
+            image_data,
+            offset: position,
+            c: PhantomData,
+        }
+    }
+}
+
+impl<I, C> Transform for Image<'_, I, C> {
+    /// Translate the image by a given delta, returning a new image
+    ///
+    /// # Examples
+    ///
+    /// ## Move an image around
+    ///
+    /// This examples moves a 4x4 black and white image by `(10, 20)` pixels without mutating the
+    /// original image
+    ///
+    /// ```rust
+    /// use embedded_graphics::{
+    ///     geometry::Point,
+    ///     image::{Image, ImageRaw},
+    ///     pixelcolor::BinaryColor,
+    ///     prelude::*,
+    /// };
+    ///
+    /// let image: ImageRaw<BinaryColor> = ImageRaw::new(&[0xff, 0x00, 0xff, 0x00], 4, 4);
+    ///
+    /// let image = Image::new(&image, Point::zero());
+    ///
+    /// let image_moved = image.translate(Point::new(10, 20));
+    ///
+    /// assert_eq!(image.top_left(), Point::zero());
+    /// assert_eq!(image_moved.top_left(), Point::new(10, 20));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            image_data: self.image_data,
+            offset: self.offset + by,
+            c: PhantomData,
+        }
+    }
+
+    /// Translate the image by a given delta, modifying the original object
+    ///
+    /// # Examples
+    ///
+    /// ## Move an image around
+    ///
+    /// This examples moves a 4x4 black and white image by `(10, 20)` pixels by mutating the
+    /// original image
+    ///
+    /// ```rust
+    /// use embedded_graphics::{
+    ///     geometry::Point,
+    ///     image::{Image, ImageRaw},
+    ///     pixelcolor::BinaryColor,
+    ///     prelude::*,
+    /// };
+    ///
+    /// let image: ImageRaw<BinaryColor> = ImageRaw::new(&[0xff, 0x00, 0xff, 0x00], 4, 4);
+    ///
+    /// let mut image = Image::new(&image, Point::zero());
+    ///
+    /// image.translate_mut(Point::new(10, 20));
+    ///
+    /// assert_eq!(image.top_left(), Point::new(10, 20));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.offset += by;
+
+        self
+    }
+}
+
+impl<'a, 'b, I, C> Drawable<C> for &'a Image<'b, I, C>
+where
+    &'b I: IntoPixelIter<C>,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_iter(
```

Comment:
I can't remember if I tried this when fixing the lifetimes, but can this use the `ImageIterator` defined below to reduce code duplication?

Reply:
Yes it can, good catch :)